src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -111,6 +116,9 @@ void GcsJobManager::HandleAddJob(rpc::AddJobRequest request,
       gcs_table_storage_->JobTable().Put(job_id, mutable_job_table_data, on_done);
   if (!status.ok()) {
     on_done(status);
+  } else {
+    const bool insert_suc = running_job_ids_.insert(job_id).second;
+    RAY_CHECK(insert_suc) << job_id.Hex() << " already inserted.";
```

Comment:
remove this. The `Put()` is designed so that if `status.ok()`, it calls `on_done`; else it does not. Here we just complete it by also calling `on_done` even if persist failed. I don't think we need to check here. Actually `AddJob` may be idempotent, @rkooo567 please confirm?

Reply:
Understand, I guess one possible reason is client-side retry. Return value check removed.